Mathlib/LinearAlgebra/RootSystem/RootPositive.lean

```diff
@@ -44,14 +42,73 @@ variable {ι R S M N : Type*} [LinearOrderedCommRing S] [CommRing R] [Algebra S
 
 namespace RootPairing
 
-lemma two_mul_apply_root_root_of_isOrthogonal (P : RootPairing ι R M N)
-    (B : LinearMap.BilinForm R M) (i j : ι) (h : B.IsOrthogonal (P.reflection j)) :
-    2 * B (P.root i) (P.root j) = P.pairing i j * B (P.root j) (P.root j) := by
+/-- Given a root pairing, this is an invariant symmetric bilinear form. -/
+structure InvariantForm (P : RootPairing ι R M N) where
+  /-- The bilinear form bundled inside an `InvariantForm`. -/
+  form : LinearMap.BilinForm R M
+  symm : form.IsSymm
+  ne_zero (i : ι) : form (P.root i) (P.root i) ≠ 0
+  isOrthogonal_reflection (i : ι) : form.IsOrthogonal (P.reflection i)
+
+namespace InvariantForm
+
+variable {P : RootPairing ι R M N} (B : P.InvariantForm) (i j : ι)
+
+lemma apply_root_ne_zero : B.form (P.root i) ≠ 0 :=
+  fun contra ↦ B.ne_zero i <| by simp [contra]
+
+lemma two_mul_apply_root_root :
+    2 * B.form (P.root i) (P.root j) = P.pairing i j * B.form (P.root j) (P.root j) := by
   rw [two_mul, ← eq_sub_iff_add_eq]
-  nth_rw 1 [← h]
+  nth_rw 1 [← B.isOrthogonal_reflection j]
   rw [reflection_apply, reflection_apply_self, root_coroot'_eq_pairing, LinearMap.map_sub₂,
     LinearMap.map_smul₂, smul_eq_mul, LinearMap.map_neg, LinearMap.map_neg, mul_neg, neg_sub_neg]
 
+lemma pairing_mul_eq_pairing_mul_swap :
+    P.pairing j i * B.form (P.root i) (P.root i) =
+    P.pairing i j * B.form (P.root j) (P.root j) := by
+  rw [← B.two_mul_apply_root_root i j, ← B.two_mul_apply_root_root j i, ← B.symm.eq,
+    RingHom.id_apply]
+
+@[simp]
+lemma apply_reflection_reflection (x y : M) :
+    B.form (P.reflection i x) (P.reflection i y) = B.form x y :=
+  B.isOrthogonal_reflection i x y
+
+@[simp]
+lemma rootForm_weylGroup_apply (g : P.weylGroup) (x y : M) :
```

Comment:
Other lemmas of the form `B.form _ _` are named with `apply.*`. Should this be
```suggestion
lemma apply_weylGroup_smul (g : P.weylGroup) (x y : M) :
```
or something like that?

Reply:
Agreed, yes.